test/k8sT/DatapathConfiguration.go

```diff
@@ -481,6 +481,52 @@ var _ = Describe("K8sDatapathConfig", func() {
 		})
 	})
 
+	SkipContextIf(helpers.DoesNotRunOnNetNextKernel, "Wireguard encryption", func() {
+		It("Pod2pod is encrypted", func() {
+			deploymentManager.DeployCilium(map[string]string{
+				"tunnel":               "disabled",
+				"autoDirectNodeRoutes": "true",
+				"wireguard.enabled":    "true",
+				"l7Proxy":              "false",
+			}, DeployCiliumOptionsAndDNS)
+
+			randomNamespace := deploymentManager.DeployRandomNamespaceShared(DemoDaemonSet)
+			deploymentManager.WaitUntilReady()
```

Comment:
Shouldn't this be extracted out into a `BeforeAll`? In case new `It()` are added, it would make sense to move the setup of the test outside of the actual test.

Reply:
I'm planning to add one test for a different configuration (with tunneling enabled once it's resolved). Therefore, I moved the deployment to `It()`.